internal/manifest/version_edit.go

```diff
@@ -451,11 +453,14 @@ func (b *BulkVersionEdit) Accumulate(ve *VersionEdit) {
 		// VersionEdit at the same level (though files can move across levels).
 		if dmap := b.Deleted[nf.Level]; dmap != nil {
 			if _, ok := dmap[nf.Meta.FileNum]; ok {
-				panic(fmt.Sprintf("file deleted %d before it was inserted\n", nf.Meta.FileNum))
+				return errors2.CorruptionError{
```

Comment:
Does this classify as corruption? When called from `versionSet.load` it does, but not when from `versionSet.logAndApply`. The latter would be a logic error in compaction.